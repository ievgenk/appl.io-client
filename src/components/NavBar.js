import React from "react";
import styled from "styled-components";

const MainNav = styled.nav`
  border-bottom: 3px solid black;
  grid-column: 1/-1;
  font-size: ${props => (props.landing ? "2.5rem" : "2rem")};
  padding-left: 20px;
  display: flex;
  align-items: center;

  h1 {
    margin: 0;
  }
`;

export default () => {
  return (
    <MainNav>
      <h1>Appl.io</h1>
    </MainNav>
  );
};
