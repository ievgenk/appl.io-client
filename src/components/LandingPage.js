import React, { Component } from "react";
import NavBar from "./NavBar";
import LoginForm from "./LoginForm";
import RegisterForm from "./RegisterForm";
import styled from "styled-components";
import { BrowserRouter as Router, Route, Redirect } from "react-router-dom";

const LandingGrid = styled.div`
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  grid-gap: 10px;
  height: 100vh;
  width: 100vw;
  grid-template-rows: repeat(10, minmax(100px, max-content));
`;

const LandingMessage = styled.h1`
  grid-column: 2 / 6;
  font-size: 5.5rem;
  grid-row: 3 / 5;
`;

export default class LandingPage extends Component {
  render() {
    return !this.props.isLoggedIn ? (
      <Router>
        <LandingGrid>
          <NavBar />
          <Route exact path="/" component={LoginForm} />
          <Route path="/register" component={RegisterForm} />
          <LandingMessage>
            🔍 Track Your
            <br /> 💼 Job
            <br /> 📝 Applications
            <br /> 👌 With Ease!
          </LandingMessage>
        </LandingGrid>
      </Router>
    ) : (
      <Redirect to="/dashboard" />
    );
  }
}

LandingPage.defaultProps = {
  isLoggedIn: false
};
