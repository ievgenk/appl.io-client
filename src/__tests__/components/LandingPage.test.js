import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store"; // Smart components
import LandingPage from "../../components/LandingPage";
import NavBar from "../../components/NavBar";

describe("<LandingPage />", () => {
  it("Render()", () => {
    const component = shallow(<LandingPage />);
    expect(
      component.contains(
        <h1>
          <span role="img" aria-label="maginfying glass emoji">
            🔍
          </span>
          Track Your
          <br />
          <span role="img" aria-label="suitcase emoji">
            💼
          </span>
          Job
          <br />
          <span
            role="img"
            aria-label="document with pencil writing on it emoji"
          >
            📝
          </span>
          Applications
          <br />
          <span role="img" aria-label="ok hand gesture emoji">
            👌
          </span>
          With Ease!
        </h1>
      )
    ).toBeTruthy();
    expect(component.contains(<NavBar position="landing" />)).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});
