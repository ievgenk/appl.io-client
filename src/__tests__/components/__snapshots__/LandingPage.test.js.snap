// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage /> Render() 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LandingPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Connect(NavBar)
          position="landing"
        />,
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/register"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
        </Switch>,
        <styled.span>
          <h1>
            <span
              aria-label="maginfying glass emoji"
              role="img"
            >
              üîç
            </span>
            Track Your
            <br />
            <span
              aria-label="suitcase emoji"
              role="img"
            >
              üíº
            </span>
            Job
            <br />
            <span
              aria-label="document with pencil writing on it emoji"
              role="img"
            >
              üìù
            </span>
            Applications
            <br />
            <span
              aria-label="ok hand gesture emoji"
              role="img"
            >
              üëå
            </span>
            With Ease!
          </h1>
        </styled.span>,
        <ToastContainer
          autoClose={2000}
          bodyClassName={null}
          className={null}
          closeButton={
            <CloseButton
              ariaLabel="close"
            />
          }
          closeOnClick={true}
          draggable={true}
          draggablePercent={80}
          hideProgressBar={false}
          newestOnTop={false}
          pauseOnFocusLoss={true}
          pauseOnHover={true}
          position="top-right"
          progressClassName={null}
          rtl={false}
          style={null}
          toastClassName={null}
          transition={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "position": "landing",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/register"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/register",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <h1>
            <span
              aria-label="maginfying glass emoji"
              role="img"
            >
              üîç
            </span>
            Track Your
            <br />
            <span
              aria-label="suitcase emoji"
              role="img"
            >
              üíº
            </span>
            Job
            <br />
            <span
              aria-label="document with pencil writing on it emoji"
              role="img"
            >
              üìù
            </span>
            Applications
            <br />
            <span
              aria-label="ok hand gesture emoji"
              role="img"
            >
              üëå
            </span>
            With Ease!
          </h1>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                aria-label="maginfying glass emoji"
                role="img"
              >
                üîç
              </span>,
              "Track Your",
              <br />,
              <span
                aria-label="suitcase emoji"
                role="img"
              >
                üíº
              </span>,
              "Job",
              <br />,
              <span
                aria-label="document with pencil writing on it emoji"
                role="img"
              >
                üìù
              </span>,
              "Applications",
              <br />,
              <span
                aria-label="ok hand gesture emoji"
                role="img"
              >
                üëå
              </span>,
              "With Ease!",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "maginfying glass emoji",
                "children": "üîç",
                "role": "img",
              },
              "ref": null,
              "rendered": "üîç",
              "type": "span",
            },
            "Track Your",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "suitcase emoji",
                "children": "üíº",
                "role": "img",
              },
              "ref": null,
              "rendered": "üíº",
              "type": "span",
            },
            "Job",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "document with pencil writing on it emoji",
                "children": "üìù",
                "role": "img",
              },
              "ref": null,
              "rendered": "üìù",
              "type": "span",
            },
            "Applications",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "ok hand gesture emoji",
                "children": "üëå",
                "role": "img",
              },
              "ref": null,
              "rendered": "üëå",
              "type": "span",
            },
            "With Ease!",
          ],
          "type": "h1",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoClose": 2000,
          "bodyClassName": null,
          "className": null,
          "closeButton": <CloseButton
            ariaLabel="close"
          />,
          "closeOnClick": true,
          "draggable": true,
          "draggablePercent": 80,
          "hideProgressBar": false,
          "newestOnTop": false,
          "pauseOnFocusLoss": true,
          "pauseOnHover": true,
          "position": "top-right",
          "progressClassName": null,
          "rtl": false,
          "style": null,
          "toastClassName": null,
          "transition": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Connect(NavBar)
            position="landing"
          />,
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/register"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
          </Switch>,
          <styled.span>
            <h1>
              <span
                aria-label="maginfying glass emoji"
                role="img"
              >
                üîç
              </span>
              Track Your
              <br />
              <span
                aria-label="suitcase emoji"
                role="img"
              >
                üíº
              </span>
              Job
              <br />
              <span
                aria-label="document with pencil writing on it emoji"
                role="img"
              >
                üìù
              </span>
              Applications
              <br />
              <span
                aria-label="ok hand gesture emoji"
                role="img"
              >
                üëå
              </span>
              With Ease!
            </h1>
          </styled.span>,
          <ToastContainer
            autoClose={2000}
            bodyClassName={null}
            className={null}
            closeButton={
              <CloseButton
                ariaLabel="close"
              />
            }
            closeOnClick={true}
            draggable={true}
            draggablePercent={80}
            hideProgressBar={false}
            newestOnTop={false}
            pauseOnFocusLoss={true}
            pauseOnHover={true}
            position="top-right"
            progressClassName={null}
            rtl={false}
            style={null}
            toastClassName={null}
            transition={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "position": "landing",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/register"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/register",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <h1>
              <span
                aria-label="maginfying glass emoji"
                role="img"
              >
                üîç
              </span>
              Track Your
              <br />
              <span
                aria-label="suitcase emoji"
                role="img"
              >
                üíº
              </span>
              Job
              <br />
              <span
                aria-label="document with pencil writing on it emoji"
                role="img"
              >
                üìù
              </span>
              Applications
              <br />
              <span
                aria-label="ok hand gesture emoji"
                role="img"
              >
                üëå
              </span>
              With Ease!
            </h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  aria-label="maginfying glass emoji"
                  role="img"
                >
                  üîç
                </span>,
                "Track Your",
                <br />,
                <span
                  aria-label="suitcase emoji"
                  role="img"
                >
                  üíº
                </span>,
                "Job",
                <br />,
                <span
                  aria-label="document with pencil writing on it emoji"
                  role="img"
                >
                  üìù
                </span>,
                "Applications",
                <br />,
                <span
                  aria-label="ok hand gesture emoji"
                  role="img"
                >
                  üëå
                </span>,
                "With Ease!",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-label": "maginfying glass emoji",
                  "children": "üîç",
                  "role": "img",
                },
                "ref": null,
                "rendered": "üîç",
                "type": "span",
              },
              "Track Your",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-label": "suitcase emoji",
                  "children": "üíº",
                  "role": "img",
                },
                "ref": null,
                "rendered": "üíº",
                "type": "span",
              },
              "Job",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-label": "document with pencil writing on it emoji",
                  "children": "üìù",
                  "role": "img",
                },
                "ref": null,
                "rendered": "üìù",
                "type": "span",
              },
              "Applications",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-label": "ok hand gesture emoji",
                  "children": "üëå",
                  "role": "img",
                },
                "ref": null,
                "rendered": "üëå",
                "type": "span",
              },
              "With Ease!",
            ],
            "type": "h1",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoClose": 2000,
            "bodyClassName": null,
            "className": null,
            "closeButton": <CloseButton
              ariaLabel="close"
            />,
            "closeOnClick": true,
            "draggable": true,
            "draggablePercent": 80,
            "hideProgressBar": false,
            "newestOnTop": false,
            "pauseOnFocusLoss": true,
            "pauseOnHover": true,
            "position": "top-right",
            "progressClassName": null,
            "rtl": false,
            "style": null,
            "toastClassName": null,
            "transition": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
